smqtk_classifier/impls/classify_descriptor_supervised/sklearn_svm.py

```diff
@@ -0,0 +1,449 @@
+                                     probability=probability_map[params['-b']],
+                                     class_weight=weights)
+        elif self.svm_type == 'nu-SVC':
+            self.svm_model = svm.NuSVC(kernel=kernel_map[params['-t']],
+                                       probability=probability_map[params['-b']])
+        elif self.svm_type == 'one-class SVM':
+            self.svm_model = svm.OneClassSVM(kernel=kernel_map[params['-t']])
+        elif self.svm_type == 'epsilon-SVR':
+            self.svm_model = svm.SVR(C=self.train_params['-c'],
+                                     kernel=kernel_map[params['-t']])
+        elif self.svm_type == 'nu-SVR':
+            self.svm_model = svm.NuSVR(C=self.train_params['-c'],
+                                       kernel=kernel_map[params['-t']])
+
+        LOG.debug("Training SVM model")
+        self.svm_model.fit(train_vectors, train_labels)  # type: ignore
+        LOG.debug("Training SVM model -- Done")
+
+        if self.svm_label_map_elem and self.svm_label_map_elem.writable():
+            LOG.debug("saving labels to element (%s)", self.svm_label_map_elem)
+            self.svm_label_map_elem.set_bytes(
+                pickle.dumps(self.svm_label_map, -1)
+            )
+        if self.svm_model_elem and self.svm_model_elem.writable():
+            LOG.debug("saving model to element (%s)", self.svm_model_elem)
+            # LibSvm I/O only works with filepaths, thus the need for an
+            # intermediate temporary file.
+            fd, fp = tempfile.mkstemp()
+            try:
+                joblib.dump(self.svm_model, fp)
+                # Use the file descriptor to create the file object.
+                # This avoids reopening the file and will automatically
+                # close the file descriptor on exiting the with block.
+                # fdopen() is required because in Python 2 open() does
+                # not accept a file descriptor.
+                with os.fdopen(fd, 'rb') as f:
+                    self.svm_model_elem.set_bytes(f.read())
+            finally:
+                os.remove(fp)
+
+    def get_labels(self) -> Sequence[Hashable]:
+        if not self.has_model():
+            raise RuntimeError("No model loaded")
+        return list(self.svm_label_map.values())
+
+    def _classify_arrays(self, array_iter: Union[np.ndarray, Iterable[np.ndarray]]) -> Iterator[Dict[Hashable, float]]:
+        if not self.has_model():
+            raise RuntimeError("No SVM model present for classification")
+        assert self.svm_model is not None, (
+            "Should have an SVM model at this point."
+        )
+
+        # Dump descriptors into a matrix for normalization and use in
+        # prediction.
+        vec_mat = numpy.array(list(array_iter))
+        vec_mat = self._norm_vector(vec_mat)
+        n_jobs = self.n_jobs
+        if n_jobs is not None:
+            n_jobs = min(len(vec_mat), n_jobs)
```

Comment:
I don't think the previous parallelism model holds for the now-use of sklearn's SVC, whose predict_proba function I believe is vectorized which makes parallelism make little to no sense here.